harness/variables_test.go

```diff
@@ -0,0 +1,230 @@
+package harness
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+// Helper function to create a temporary file and set the environment variable
+func setupTestFile(t *testing.T, envVar string) string {
+	t.Helper()
+	tmpDir := t.TempDir()
+	tmpFile := filepath.Join(tmpDir, "test.env")
+	err := os.WriteFile(tmpFile, []byte(""), 0644)
+	if err != nil {
+		t.Fatalf("Failed to create temporary file: %v", err)
+	}
+	os.Setenv(envVar, tmpFile)
+	return tmpFile
+}
+
+// Helper function to create a temporary .out file and set the environment variable
+func setupOutTestFile(t *testing.T, envVar string) string {
+	t.Helper()
+	tmpDir := t.TempDir()
+	tmpFile := filepath.Join(tmpDir, "test.out")
+	err := os.WriteFile(tmpFile, []byte(""), 0644)
+	if err != nil {
+		t.Fatalf("Failed to create temporary file: %v", err)
+	}
+	os.Setenv(envVar, tmpFile)
+	return tmpFile
+}
```

Comment:
just make this generic and take ext as input as well

Reply:
Done.